Sources/URLEncodedForm/URLEncodedFormEncoder.swift

```diff
@@ -0,0 +1,268 @@
+/// Encodes `Encodable` instances to `application/x-www-form-urlencoded` data.
+///
+///     print(user) /// User
+///     let data = try URLEncodedFormEncoder().encode(user)
+///     print(data) /// Data
+///
+/// URL-encoded forms are commonly used by websites to send form data via POST requests. This encoding is relatively
+/// efficient for small amounts of data but must be percent-encoded.  `multipart/form-data` is more efficient for sending
+/// large data blobs like files.
+///
+/// See [Mozilla's](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST) docs for more information about
+/// url-encoded forms.
+public struct URLEncodedFormEncoder {
+    /// Create a new `URLEncodedFormEncoder`.
+    public init() {}
+
+    /// Encodes the supplied `Encodable` object to `Data`.
+    ///
+    ///     print(user) // User
+    ///     let data = try URLEncodedFormEncoder().encode(user)
+    ///     print(data) // "name=Vapor&age=3"
+    ///
+    /// - parameters:
+    ///     - encodable: Generic `Encodable` object (`E`) to encode.
+    /// - returns: Encoded `Data`
+    /// - throws: Any error that may occur while attempting to encode the specified type.
+    public func encode<E>(_ encodable: E) throws -> String
+        where E: Encodable
+    {
+        let encoder = _Encoder(codingPath: [])
+        try encodable.encode(to: encoder)
+        let serializer = URLEncodedFormSerializer()
+        guard case .dictionary(let dict) = encoder.container!.data!.resolve() else {
+            throw URLEncodedFormError(
+                identifier: "invalidTopLevel",
+                reason: "form-urlencoded requires a top level dictionary"
+            )
+        }
+        print(dict)
```

Comment:
A debug print is making it into release builds.

```suggestion
        #if DEBUG
             print(dict)
        #endif
```